core/src/main/scala/cats/data/StateT.scala

```diff
@@ -12,25 +12,32 @@ import cats.syntax.either._
 final class StateT[F[_], S, A](val runF: F[S => F[(S, A)]]) extends Serializable {
 
   def flatMap[B](fas: A => StateT[F, S, B])(implicit F: Monad[F]): StateT[F, S, B] =
```

Comment:
this only needs `FlatMap[F]` in this formulation, but I didn't want to change binary compatibility for a negligible gain.